### PR TITLE
Fix test_default_route.py for t2_min topology

### PR DESCRIPTION
**Pull request by abdosi**
What  I did:

- Handle the route parsing logic to get source ip to handle the case when there is no ECMP involved.
- Update uplink_ns to take care of only upstream neighbors that advertise default route.

Why I did:
Without this change for packet-chassis t2_min topology this test case was failing.

How I verify:

Manual Verification

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/common/devices/sonic.py<br/>Fixing tests/route/test_default_route.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/common/devices/sonic.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wenyiz2021** [comment]
hi @abdosi could you update your test result as well?

**Review by wenyiz2021 — APPROVED** [review]

**Comment by wenyiz2021** [comment]
is this for t2-min topo only?

File: tests/route/test_default_route.py [hunk 1]
```diff
@@ -176,7 +178,7 @@ def test_default_route_with_bgp_flap(duthosts, tbinfo):
     # Get uplink namespaces/asics for multi-asic
     if duthost.is_multi_asic:
         bgp_name_to_ns_mapping = duthost.get_bgp_name_to_ns_mapping()
-        uplink_ns = get_uplink_ns(tbinfo, bgp_name_to_ns_mapping)
+        uplink_ns = get_uplink_ns(tbinfo, bgp_name_to_ns_mapping, config_facts['DEVICE_NEIGHBOR_METADATA'])
```
Comment (wenyiz2021) [inline tests/route/test_default_route.py:181]: `def get_uplink_ns` only has 2 arguments, should we add a default argument in the function definition?

**Pull request merged**